src/test/system_tests.cpp

```diff
@@ -29,25 +29,44 @@ BOOST_AUTO_TEST_CASE(dummy)
 
 BOOST_AUTO_TEST_CASE(run_command)
 {
+#ifdef WIN32
+    // https://www.winehq.org/pipermail/wine-devel/2008-September/069387.html
+    auto hntdll = GetModuleHandleA("ntdll.dll");
+    assert(hntdll);
+    const bool wine_runtime = GetProcAddress(hntdll, "wine_get_version");
```

Comment:
This check will fail on Windows. I'm guessing that's expected but just checking?

Reply:
The goal is to detect that binaries, which are cross-compiled for Windows, are running in Wine.

Reply:
Does that mean this PR is intended to be a fix for Windows or for wine?

Reply:
For both. Wine is used to run test suite on Linux after cross-compiling for Windows. This need root as Wine and Windows runtime produce different error messages: https://github.com/bitcoin/bitcoin/blob/db73ac8c517f91ed5175b3635eb37ed56852ff91/src/test/system_tests.cpp#L80

Reply:
FWIW, the same behaviour was before https://github.com/bitcoin/bitcoin/pull/29489.